src/main/java/core/basesyntax/Square.java

```diff
@@ -0,0 +1,21 @@
+package core.basesyntax;
+
+public class Square extends Figure {
+    private double side;
+
+    public Square(String color, double side) {
+        super(color);
+        this.side = side;
+    }
+
+    @Override
+    public double getArea() {
+        return side * side;
+    }
+
+    @Override
+    public void draw() {
+        System.out.println("Figure: square, area: " + getArea() + " sq. units, side: " + side + " units, color: " + color);
```

Comment:
The `color` variable is protected in the `Figure` class, so it is not directly accessible in the `Square` class. You need to access it through a getter method or directly from the superclass. Consider adding a `getColor()` method in the `Figure` class to retrieve the color.